qiskit/algorithms/optimizers/gradient_descent.py

```diff
@@ -119,18 +183,32 @@ def __init__(
                 updates. See the docstring for an example.
             tol: If the norm of the parameter update is smaller than this threshold, the
                 optimizer is converged.
-            perturbation: If no gradient is passed to ``GradientDescent.optimize`` the gradient is
+            perturbation: If no gradient is passed to :meth:`~.minimize` the gradient is
                 approximated with a symmetric finite difference scheme with ``perturbation``
                 perturbation in both directions (defaults to 1e-2 if required).
-                Ignored if a gradient callable is passed to ``GradientDescent.optimize``.
+                Ignored if a gradient callable is passed to `~.minimize`.
         """
-        super().__init__()
-
-        self.maxiter = maxiter
+        super().__init__(maxiter=maxiter)
+        self.callback = callback
```

Comment:
This is a public instance var? I did not see setter/getter properties - but then I guess it was like that before and moved up and there are other public instance vars there too. Always the issue with doing that versus properties was documentation - these do not seem to be documented in the current Terra API ref.

Reply:
Good point. I think in that case we should make them attributes.